### PR TITLE
Defer wait on startOvnKube might prevent panic crash

### PR DESCRIPTION
**Pull request by jcaamano**
On goroutine panic, defers are executed. If we wait there for something
that will never happen, we prevent the process to fully crash. Make sure
we cancel the context before waiting.

### CODE REVIEW

**Comment by kyrtapz** [comment]
/lgtm

This issue would happen on any early exit, not only on panics.

**Comment by jcaamano** [comment]
> /lgtm
> 
> This issue would happen on any early exit, not only on panics.

That made me think that we are not stopping the metrics server gracefully then. Going for a different approach.

File: go-controller/cmd/ovnkube/ovnkube.go [hunk 1]
```diff
@@ -283,14 +283,19 @@ func startOvnKube(ctx *cli.Context, cancel context.CancelFunc) error {
 
 	eventRecorder := util.EventRecorder(ovnClientset.KubeClient)
 
-	wgMetricSrv := &sync.WaitGroup{}
+	ovnKubeStartWg := &sync.WaitGroup{}
+	defer func() {
```
Comment (flavio-fernandes) [inline go-controller/cmd/ovnkube/ovnkube.go:287]: better!
/lgtm

**Pull request merged**